--- a/pyn/nifdefs.py
+++ b/pyn/nifdefs.py
@@ -92,7 +92,8 @@
     bhkMoppBvTreeShapeBufType = 69
     bhkPackedNiTriStripsShapeBufType = 70
     bhkCompressedMeshShapeBufType = 71
-    COUNT = 72
+    BSDecalPlacementVectorExtraDataBufType = 72
+    COUNT = 73
 
 
 class NiShaderBuf(pynStructure):
@@ -486,6 +487,7 @@
         ('bufSize', c_uint16),
         ('bufType', c_uint16),
         ('shapeID', c_uint32),
+        ('buildType', c_uint8),
     ]
     def __init__(self, values=None):
         super().__init__(values=values)
@@ -509,6 +511,15 @@
         ('bufType', c_uint16),
         ('radius', c_float),
         ('dataID', c_uint32),
+        ('bitsPerIndex', c_uint32),
+        ('bitsPerWIndex', c_uint32),
+        ('maskIndex', c_uint32),
+        ('maskWIndex', c_uint32),
+        ('error', c_float),
+        ('materialType', c_uint8),
+        ('userData', c_uint32),
+        ('unkFloat', c_float),
+        ('targetID', c_uint32),
     ]
     def __init__(self, values=None):
         super().__init__(values=values)
@@ -818,6 +829,24 @@
                     self.entry_points |= int(e, 0)
 
 
+class BSDecalPlacementVectorExtraDataBuf(pynStructure):
+    _fields_ = [
+        ('bufSize', c_uint16),
+        ('bufType', c_uint16),
+        ("nameID", c_uint32),
+        ("numVectorBlocks", c_uint16) ]
+
+    def __init__(self, values=None):
+        super().__init__(values=values)
+        self.bufType = PynBufferTypes.BSDecalPlacementVectorExtraDataBufType
+
+
+class DecalVectorBuf(Structure):
+    _fields_ = [
+        ("point", c_float * 3),
+        ("normal", c_float * 3) ]
+
+
 class ConnectPointBuf(pynStructure):
     _fields_ = [
         ("parent", c_char * 256),
